src/plugins/signoff/components.js

```diff
@@ -0,0 +1,267 @@
+import AdminLink from "../../components/AdminLink";
+import { ProgressBar, ProgressStep } from "./ProgressBar.js";
+
+
+export default class SignoffToolBar extends React.Component {
+  props: {
+    sessionState: SessionState,
+    bucketState: BucketState,
+    collectionState: CollectionState,
+    signoff: SignoffState,
+    requestReview: () => void,
+    approveChanges: () => void,
+    declineChanges: () => void,
+  };
+
+  render() {
+    const {
+      // Global state
+      sessionState,
+      bucketState,
+      collectionState,
+      // Plugin state
+      signoff={},
+      // Actions
+      requestReview,
+      approveChanges,
+      declineChanges} = this.props;
+
+    const canEdit = canEditCollection(sessionState, bucketState, collectionState);
+
+    // The above sagas refresh the global state via `routeLoadSuccess` actions.
+    // Use the global so that the toolbar is refreshed when status changes.
+    const {data: {status}} = collectionState;
+
+    // Information loaded via this plugin.
+    const {resource} = signoff;
+    // Hide toolbar if server has not kinto-signer plugin,
+    // or if this collection is not configured to be signed.
+    if (!resource) {
+      return null;
+    }
+
+    const {source, preview, destination} = resource;
+
+    // Default status is request review
+    const step = status == "to-review" ? 1 : status == "signed" ? 2 : 0;
+    return (
+      <ProgressBar>
+        <WorkInProgress label="Work in progress"
+                        step={0}
+                        currentStep={step}
+                        canEdit={canEdit}
+                        requestReview={requestReview}
+                        source={source} />
+        <Review label="Waiting review"
+                step={1}
+                currentStep={step}
+                canEdit={canEdit}
+                approveChanges={approveChanges}
+                declineChanges={declineChanges}
+                source={source}
+                preview={preview} />
+        <Signed label="Signed"
+                step={2}
+                currentStep={step}
+                canEdit={canEdit}
+                reSign={approveChanges}
+                source={source}
+                destination={destination} />
+      </ProgressBar>
+    );
+  }
+}
+
+function Date({date} : {date: number}) {
+  return <span title={humanDate(date)}>{timeago(date)}</span>;
+}
+
+//
+// Work in progress
+//
+
+function WorkInProgress({label,
+                         canEdit,
+                         currentStep,
+                         step,
+                         requestReview,
+                         source} : {label: string,
+                                    canEdit: boolean,
+                                    currentStep: number,
+                                    step: number,
+                                    requestReview: () => void,
+                                    source: SourceInfo}) {
```

Comment:
Huh. Definitely put this type definition in some dedicated type along the component declaration :)